include/RAJA/policy/openmp_target/reduce.hpp

```diff
@@ -254,17 +268,29 @@ struct TargetReduceLoc
 {
   TargetReduceLoc() = delete;
   TargetReduceLoc(const TargetReduceLoc &) = default;
-  explicit TargetReduceLoc(T init_val, IndexType init_loc)
+  explicit TargetReduceLoc(T init_val_, IndexType init_loc,
+                           T identity_ = Reducer::identity)
       : info(),
-        val(Reducer::identity, Reducer::identity, info),
+        val(identity_, identity_, info),
         loc(init_loc, IndexType(RAJA::reduce::detail::DefaultLoc<IndexType>().value()), info),
-        initVal(init_val),
-        finalVal(Reducer::identity),
+        initVal(init_val_),
+        finalVal(identity_),
         initLoc(init_loc),
         finalLoc(IndexType(RAJA::reduce::detail::DefaultLoc<IndexType>().value()))
   {
   }
 
+  void reset(T init_val_, T identity_ = Reducer::identity)
```

Comment:
I still don't see init_loc here.

Reply:
Ah I see, init_loc in reset is not supported with the other back-ends. Shall we add that in this PR? 
It might just be as simple as updating it under the base BaseReduceMinLoc/BaseReduceMaxLoc used to instantiate all other back-ends.

Reply:
Probably good to add those here. The reset methods should be adequately covered in the reduce-sanity tests I added recently and the reducer reset unit tests Robert added. Please let me know if you run into any issues. Thanks.

Reply:
If we add this, let's make it a defaulted parameter, so that the API doesn't change. I was looking for instances of reset() being used in Kull, but couldn't find any. However, Riyaz expressed some concern to me about changing the reduceloc API, so we should keep it as similar to before as possible.

Reply:
Sure I can do that here.